pkg/kudoctl/cmd/verify/verify.go

```diff
@@ -4,49 +4,38 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
 	"github.com/kudobuilder/kudo/pkg/kudoctl/packages"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/template"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/packages/verifier"
 )
 
-type ParamWarning string
-type ParamWarnings []ParamWarning
-type ParamError string
-type ParamErrors []ParamError
-
-var verifiers = []ParameterVerifier{
+var verifiers = []verifier.PackageVerifier{
 	DuplicateVerifier{},
 	InvalidCharVerifier{";,"},
+	template.ParametersVerifier{},
+	template.ReferenceVerifier{},
 }
 
-// Parameters verifies parameters
-func Parameters(params packages.Parameter) (warnings ParamWarnings, errors ParamErrors) {
+// PackageFiles verifies operator package files
+func PackageFiles(pf *packages.Files) (warnings verifier.ParamWarnings, errors verifier.ParamErrors) {
 	for _, verifier := range verifiers {
-		w, err := verifier.Verify(params)
-		warnings = append(warnings, w...)
-		errors = append(errors, err...)
+		ws, errs := verifier.Verify(pf)
+		warnings = append(warnings, ws...)
+		errors = append(errors, errs...)
 	}
 	return warnings, errors
 }
 
-// ParameterVerifier defines the interface for all parameter verifiers
-type ParameterVerifier interface {
-	Verify(params packages.Parameter) (ParamWarnings, ParamErrors)
-}
-
-func CreateParamError(param v1beta1.Parameter, reason string) ParamError {
-	return ParamError(fmt.Sprintf("parameter %q %s", param.Name, reason))
-}
-
 // DuplicateVerifier provides verification that there are no duplicates disallowing casing (Kudo and kudo are duplicates)
 type DuplicateVerifier struct {
 }
 
-func (DuplicateVerifier) Verify(params packages.Parameter) (warnings ParamWarnings, errors ParamErrors) {
+func (DuplicateVerifier) Verify(pf *packages.Files) (warnings verifier.ParamWarnings, errors verifier.ParamErrors) {
```

Comment:
should we move these to verify package as well? would be nice to have all the verifiers in one package or do you see any advantage of keeping these separate?

Reply:
to get to the level of encapsulation @zen-dog requested which current makes sense to me as well... it is necessary for verifiers to be in different packages.   The template verifiers for instance are under `templates` package.   To put all verifiers in a verifiers package would request exposing template details... same for other verifiers.